src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -342,6 +349,17 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
             factoryContext.TrackedParameters.Add(parameter.Name, RequestDelegateFactoryConstants.RouteOrQueryStringParameter);
             return BindParameterFromRouteValueOrQueryString(parameter, parameter.Name, factoryContext);
         }
+        else if (parameter.ParameterType.IsArray && factoryContext.DisableInferredFromBody && ParameterBindingMethodCache.HasTryParseMethod(parameter.ParameterType.GetElementType()!))
+        {
+            // We only infer parameter types if you have an array of TryParsables and:
+            // - DisableInferredFromBody is true
+            // - TBD: You have explicitly declared a [FromBody] so it's no longer ambiguous
```

Comment:
Not sure about this but its an idea.

Reply:
It's an interesting idea. I don't love one parameter influencing the binding source of another. I like keeping the binding source inference rules as simple as possible.

Reply:
Ya, the more I thought about the more I think we should keep the rules simple. Implementing it would require changing how we bound the arguments doing code generation in a second pass after looking at all of the arguments.